finite_difference/src/field_mod.f90

```diff
@@ -1009,7 +1009,7 @@ function fld_checksum(field) result(val)
     type(r2d_field), intent(in) :: field
     real(go_wp) :: val
 
-    val = array_checksum(field%data, field%data_on_device,           &
+    val = array_checksum(field%get_data(), field%data_on_device,           &
```

Comment:
Please could you delete the commented-out code on lines 1017-1028 as it is no longer necessary.

Reply:
Please could you also update the docstring to say that if the data for the field is on a remote device then it is fetched first.

Reply:
Done